projects/hr-portal/src/app/hr-portal/hr-portal.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect, useCallback, useRef } from 'react';
+import './hr-portal.scss';
+
+import {
+  IgrTreeGrid,
+  IgrGridToolbar,
+  IgrGridToolbarTitle,
+  IgrGridToolbarActions,
+  IgrGridToolbarAdvancedFiltering,
+  IgrGridToolbarHiding,
+  IgrGridToolbarPinning,
+  IgrGridToolbarExporter,
+  IgrColumn,
+  IgrPaginator,
+  IgrCellTemplateContext,
+} from 'igniteui-react-grids';
+import { IgrIcon, IgrAvatar, IgrIconButton, IgrButton, registerIcon } from 'igniteui-react';
+import 'igniteui-react-grids/grids/combined';
+import 'igniteui-react-grids/grids/themes/light/fluent.css';
+import { DataService } from './../services/data.service';
+import { icons } from '../data/icons/Icons'
+
+export default function HRPortal() {
+  const [data, setData] = useState<any[]>([]);
+  const [isSorted, setIsSorted] = useState(false);
+  const dataService = React.useMemo(() => new DataService(), []);
+
+  const gridRef = useRef<IgrTreeGrid | null>(null);
+
+  useEffect(() => {
+    dataService.fetchData().then((fetchedData: any) => {
+      setData(fetchedData);
+    });
+
+    icons.forEach((element: { name: string; path: string; category: string }) => {
+      registerIcon(element.name, element.path, element.category);
+    });
+  }, [dataService]);
+
+  const avatarTemplate = (props: { dataContext: IgrCellTemplateContext }) => {
```

Comment:
You can skip the dataContext here.
`dataContext is deprecated and template context props are now available as the root object and can be accessed directly`
Please do for all templates.